pkg/manifests/manifests.go

```diff
@@ -2898,6 +2899,26 @@ func (f *Factory) ControlPlanePrometheusRule() (*monv1.PrometheusRule, error) {
 		}
 		r.Spec.Groups = groups
 	}
+	if !f.infrastructure.HighlyAvailableInfrastructure() || !f.infrastructure.HighlyAvailableControlPlane() {
+		// Aggregated API may go down at times since there is no high
+		// availability for these components. Examples of these situations may
+		// include upgrades or certificate rotations. Since we do expect downtime but
+		// we still want to be alerted in case unavailability runs for too long, we
+		// extend the time for the alert to fire.
+		for i := range r.Spec.Groups {
+			group := &r.Spec.Groups[i]
+			if group.Name == "kubernetes-system-apiserver" {
+				for j := range group.Rules {
+					rule := &group.Rules[j]
+					if rule.Alert == "KubeAggregatedAPIDown" {
+						rule.For = "15m"
+						break
+					}
+				}
+				break
+			}
+		}
```

Comment:
@simonpasquier I don't think we need this change to be SNO specific, waiting 15 minutes instead of 5 for a warning level alert should be fine for all use-cases.

If I remember correctly, in the past people from TRT/apiserver wanted this alert to be really sensitive in order to catch issues in CI, but I don't think that is the correct direction to go with. Alerts and tests should be dissociated and that's what we've tried to start doing more in origin since CI testing is relying a lot on alerts which is preventing everyone from making them more resilient and less noisy in general. So I would be in favor of making this alert more resilient in general and adding a new test in origin that would perform a more sensitive check.

Reply:
Do we know how SRE's are responding to this alert? Do we have any SLO? For SNO we can expect downtime since there is no high availability, hence you need to be a bit more tolerant, but not so sure about multi node. Is there a requirement for uptime?
If you need help for the CI count on me.

Reply:
I would ask @openshift/sre-alert-sme how they feel about the `for` duration being extended to 15 minutes from 5 minutes for the `KubeAggregatedAPIDown` alert.
And ideally this alert (as well as `KubeAggregatedAPIErrors`) should be moved to the cluster-kube-apiserver-operator project.

Reply:
No issues from me expanding this to 15m for a warning alert. Given the amount of alerts around the apiserver - this is contextual as a warning.

Reply:
Sounds like we have an agreement that setting `for` to 15m for the general case is fine.